### PR TITLE
[Dataset][s]: Update retrieval of sample and schema

### LINKED ISSUE #163: Update publisher in Portal with latest datapackage fix
We recently made a update to how data packages are generated in this issue: https://github.com/datopian/gift-publisher/issues/66, as such we need to update how returned data packages are used in Portal. 

## Acceptance
* [x] Data publisher is updated to the latest version **Done** https://github.com/datopian/gift-portal/pull/164
* [x] All code using the sample or schema property is properly updated to read from the first resource (if present) in a data package.

### PR DESCRIPTION
**Pull request by risenW**
This PR fixes #163

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -41,14 +42,13 @@ export const processMultipleRepos = async (repos) => {
...
-            data['schema'] = datapackage['resources'][0]['schema'] || []
-            data['resources'] = datapackage['resources']
-          } catch (error) {
-            data['schema'] = { fields: [] }
-            data['resources'] = datapackage['resources']
-          }
+        if ('resources' in datapackage && datapackage['resources'].length > 0) {
+          data['resources'] = datapackage['resources']
+          data['schema'] = datapackage['resources'][0]['schema'] || []
+          data['sample'] = datapackage['resources'][0]['sample']  || datapackage['sample'] || []
```
Comment (sglavoie) [inline lib/utils.js:48]: @risenW, will we have a `sample` property at the top-level somewhere or is this to account for existing datasets which might have one already, as that's what seems to be the case from line 39 https://github.com/datopian/gift-portal/pull/164/files#diff-5dfe38baf287dcf756a17c2dd63483781b53bf4b669e10efdd01e74bcd8e780aR39
? (Thanks for enlightening me :smile:.)

```suggestion
          data['sample'] = datapackage['resources'][0]['sample'] || []
```

File: lib/utils.js [hunk 2]
```diff
@@ -90,19 +90,18 @@ export const processSingleRepo = (repo) => {
...
-    if ('resources' in datapackage) {
+    if ('resources' in datapackage && datapackage['resources'].length > 0) {
       data['resources'] = datapackage['resources']
-      try {
-        data['schema'] = datapackage['resources'][0]['schema'] || []
-      } catch (error) {
-        data['schema'] = { fields: [] }
-      }
+      data['schema'] = datapackage['resources'][0]['schema'] || []
+      data['sample'] = datapackage['resources'][0]['sample']  || datapackage['sample'] || []
```
Comment (sglavoie) [inline lib/utils.js:101]: Same comment as for line 48.

**Review by sglavoie — APPROVED** [review]

File: lib/utils.js (hunk 1 above)
Comment (risenW) [reply lib/utils.js:48]: @sglavoie Yes, this is for existing datasets with sample and schema in the top level. Should we not support it?

File: lib/utils.js (hunk 2 above)
Comment (risenW) [reply lib/utils.js:101]: Same answer. If you can reupload the datasets. I can get rid of that line.

Comment (sglavoie) [reply lib/utils.js:101]: Ok, that's great, thank you! I think we can simplify the code further and make sure we don't have to care about extra properties at the top-level. I can re-upload all the datasets, that's not a problem, so we don't have to consider any existing data so far. :slightly_smiling_face:

Comment (risenW) [reply lib/utils.js:101]: Okay, I'll update the code and push again

**Review by sglavoie — APPROVED** [review]
LGTM, thank you! :+1:

**Pull request merged**